spec/requests/api/v2/users_spec.rb

```diff
@@ -48,21 +48,103 @@
     it { should have_http_status(:success) }
 
     it 'has a data section with the correct attributes' do
-      json_response = JSON.parse response.body
-      expect(json_response).to eq(
-        "data" => {
-          "id" => user.id.to_s,
-          "type" => "users",
-          "attributes" => {
-            "first_name" => user.first_name,
-            "last_name" => user.last_name,
-            "email" => user.email
-          }
-        },
-        "jsonapi" => {
-          "version" => "1.0"
-        }
+      json_response = JSON.parse(response.body)
+      data_attributes = json_response['data']['attributes']
+      expect(data_attributes['email']).to eq(user.email)
+      expect(data_attributes['first_name']).to eq(user.first_name)
+      expect(data_attributes['last_name']).to eq(user.last_name)
+    end
+  end
+
+  describe 'PATCH update' do
+    let(:params) { {} }
+
+    def perform_request
+      Timecop.freeze
+      patch(
+        api_v2_user_path(user),
+        headers: { 'HTTP_AUTHORIZATION' => credentials },
+        params: params
       )
     end
+
+    after do
+      Timecop.return
+    end
+
+    context 'with accept_terms_date_utc' do
+      let(:jsonapi_renderer) { JSONAPI::Serializable::Renderer.new }
+      let(:params) do
+        {
+          _jsonapi: jsonapi_renderer.render(
+            user,
+            class: {
+              User: API::V2::SerializableUser
+            }
+          )
+        }
+      end
+      let(:user_params)           { params.dig :_jsonapi, :data, :attributes }
+      let(:accept_terms_date_utc) { DateTime.now.utc }
+      let(:email)                 { 'dave.smith@gmail.com' }
+      let(:first_name)            { 'Dave' }
+      let(:last_name)             { 'Smith' }
+      let(:json_data)             { JSON.parse(response.body)['data'] }
+
+      before do
+        user_params.merge!(
+          email: email,
+          accept_terms_date_utc: accept_terms_date_utc,
+          first_name: first_name,
+          last_name: last_name,
+        )
+      end
+
+      subject { perform_request }
+
+      it 'updates email on the user' do
+        expect { subject }.to(change { user.reload.email }
+          .from(user.email).to(email))
+      end
+
+      it 'updates accept_terms_date_utc on the user' do
+        expect { subject }.to(change { user.reload.accept_terms_date_utc }
+          .from(user.accept_terms_date_utc).to(be_within(1.second)
+          .of(accept_terms_date_utc)))
```

Comment:
I found this indentation confusing. How about:

```
         expect { subject }.to(change { user.reload.accept_terms_date_utc }
           .from(user.accept_terms_date_utc)
           .to(be_within(1.second).of(accept_terms_date_utc)))
```

I'd even argue the other tests below and above could benefit from having the `from` and the `to` on separate lines ... my rule of thumb is that if the line is too long to chain all on one line, go all out and put each `.method` on a separate line (within reason, of course).

Reply:
Done